src/Map.jsx

```diff
@@ -75,6 +76,29 @@ class Map extends Component {
     );
   }
 
+  getStopInfo(route, stopCoordinates) {
+    let stops = this.state.selectedStops;
+    const station = this.findStationID(route, stopCoordinates);
+    const stopInfo = Object.assign({}, stopCoordinates);
+    stopInfo.sid = station.sid;
+    if (stops.length > 1) {
+      stops = [];
+    }
+    if (stops.length === 0
+      || (stops.length === 1 && stops[0] !== stopCoordinates)) {
+      console.log(`Stop Sid: ${stopInfo.sid}`);
+      stops.push(stopInfo);
+    }
+    this.setState({ selectedStops: stops });
+  }
+
+  findStationID(route, stopCoordinate) {
+    console.log(this);
```

Comment:
Interesting, I've never run into this. Searched the error and found that it's because then the method can be made `static` (https://eslint.org/docs/rules/class-methods-use-this). I'm also fine with writing `const findStationId = (route, stopCoordinate) => {};` and calling it directly.